### PR TITLE
version: add support for resolving a version from a go workspace

### PR DESCRIPTION
**Pull request by raggi**
Updates #cleanup

### CODE REVIEW

**Review by ramyfication — APPROVED** [review]

File: version/tailscale-version.sh [hunk 1]
```diff
@@ -9,11 +9,33 @@
...
+# printversion prints the tailscale version in the standard tailscale format.
+printversion() {
+	if [[ ! -f ./build_dist.sh ]]; then
+		echo >&2 "printversion must be run from the tailscale repository root"
+		exit 1
+	fi
+	eval $(./build_dist.sh shellvars)
+	git_hash=$(git rev-parse HEAD)
+	short_hash=$(echo "$git_hash" | cut -c1-9)
+	echo ${VERSION_SHORT}-t${short_hash}
```
Comment (bradfitz) [inline version/tailscale-version.sh:21]: Should there be two git hashes here?

File: version/tailscale-version.sh [hunk 2]
```diff
@@ -35,9 +57,7 @@ git clean -d -x -f
 git fetch -q --all --tags
 git checkout -q "$mod_version"
 
-eval $(./build_dist.sh shellvars)
-git_hash=$(git rev-parse HEAD)
-short_hash=$(echo "$git_hash" | cut -c1-9)
-echo ${VERSION_SHORT}-t${short_hash}
```
Comment (bradfitz) [inline version/tailscale-version.sh:41]: I guess I don't know how this historically tried to work. Ignore me for now but I'll do some digging

Comment (raggi) [reply version/tailscale-version.sh:41]: There's another append in the Makefile for the local commit. I'm working on cleaning that up, but incrementally the step right now is to open up easier diagnostic work without needing to copy/paste patches in and out.

**Comment by raggi** [comment]
Crap, this is broken on some go versions. 1.20.6:

```
% go work edit -json
{
        "Go": "1.20",
        "Use": [
                {
                        "DiskPath": ".",
                        "ModPath": "github.com/tailscale/tailscale-android"
                },
                {
                        "DiskPath": "../tailscale",
                        "ModPath": "tailscale.com"
                }
        ],
        "Replace": null
}
```

go version go1.21rc3 linux/amd64:

```
% go work edit -json
{
        "Go": "1.20",
        "Use": [
                {
                        "DiskPath": "."
                },
                {
                        "DiskPath": "../tailscale"
                }
        ],
        "Replace": null
}
```

**Pull request closed without merging**